R/load.project.R

```diff
@@ -156,9 +157,44 @@ load.project <- function(override.config = NULL)
 }
 
 
-.load.data <- function(recursive) {
+.prepare.data.ignore.regex <- function(ignore_files) {
+  ignore_files <- strsplit(ignore_files, '\\s*,\\s*')[[1]]
+  regexes <- ignore_files[grepl('^/.*/$', ignore_files)]
+  literals <- setdiff(ignore_files, regexes)
+
+  # Create regex for special characters in regex to be escaped
+  #  (welcome to backslash hell)
+  # Note that * is a regex special character but often used in literals as
+  #  wildcard
+  regex.special <- c('.', '\\', '|', '(', ')', '[', '{', '^', '$', '+', '?')
+  regex.special <- paste0('([',
+                          paste0('\\', regex.special, collapse = '|'),
+                          '])')
+  # Escape special characters in literal strings
+  literals <- gsub(regex.special, '\\\\\\1', literals)
+  # Escape wildcard * in literal strings
+  literals <- gsub('\\*', '\\.\\*', literals)
+  # Convert trailing slash to wildcard
+  literals <- gsub('/$', '/\\.\\*', literals)
+  literals <- paste0('^', literals, '$')
+
+  # Remove starting and trailing slashes from regexes
+  regexes <- gsub('(^/)|(/$)', '', regexes)
+
+  # Combine and return prepared regexes
+  paste0(c(literals, regexes), collapse = '|')
+}
+
+.load.data <- function(recursive, ignore_files = NULL) {
   .provide.directory('data')
   data.files <- dir('data', recursive = recursive)
+  ignore.index <- grepl(.prepare.data.ignore.regex(ignore_files),
+                        data.files,
+                        perl = TRUE)
+  # print(data.frame(files = data.files,
```

Comment:
whoops, I added it commented in the dreaded commit I rolled back, should have rolled back this edit as well. I actually built a new function `list.data` instead that should be called in the future so this debug statement becomes obsolete overall. That branch however is a linear descendant of this branch, so I cannot create a pull request or these edits are included twice..